src/Core/ConstraintValidator/Constraints/UniqueStateIsoCode.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\UniqueStateIsoCodeValidator;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Unique state iso code validator constraint
+ */
+class UniqueStateIsoCode extends Constraint
+{
+    /**
+     * @var string
+     */
+    public $message = 'This ISO code already exists. You cannot create two states with the same ISO code.';
+
+    /**
+     * @var int|null
+     */
+    public $excludeId = null;
```

Comment:
I'm not sure to understand this option, it would be worth adding some comment to explain its use From what I could grasp it's used to set an ID that is accepted as existing? Like when you are editing a state you need to allow its own value right?

If so it seems like this option is required to use the constraint correctly, so maybe you should add a `getRequiredOptions` (like in TypedRegexo constraint) to force setting this option no matter what, even if the `null` value is accepted

Reply:
I wonder if the option shouldn't be named `allowedId` instead since it defines a specific ID that is allowed to be used with the checked ISO code

Reply:
Commented :) 
Renamed to `excludeStateId`. 

Fixed